### PR TITLE
Validate contact form

### PR DESCRIPTION
**Pull request by fullstop125**
**Portfolio: preserve data in the browser**

In the following milestone we managed to implement the following: 
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- create a single JavaScript object with all the data from the entire form and save it in local storage.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -196,4 +196,25 @@ form.addEventListener('submit', (e) => {
...
+    name: document.getElementById('formName').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('comment').value,
+  };
+  const formObj = JSON.stringify(formDataObj);
+  localStorage.setItem('formData', formObj);
+});
+document.getElementById('reset').addEventListener('click', () => {
+  localStorage.removeItem('formData');
+});
```
Comment (cwaku) [inline script.js]: - [x] After filling out the form, and not submitting it, the entered values should be saved in the browser. Currently, the data is only saved to `localStorage` upon clicking the submit button. Kindly fix that.

File: script.js [hunk 2]
```diff
@@ -196,4 +196,25 @@ form.addEventListener('submit', (e) => {
...
+
+const formObject = localStorage.getItem('formData');
+const objectData = JSON.parse(formObject);
+document.getElementById('formName').value = objectData.name;
+document.getElementById('email').value = objectData.email;
+document.getElementById('comment').value = objectData.message;
```
Comment (cwaku) [inline script.js]: - [x] When the user loads the page, the code checks if there are any saved data in the `localstorage` and if so the input fields are pre-filled with the data. You could do this by wrapping your already written code here with an `if statement` when the page loads.
```
// For instance:

window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

**Review by cwaku — APPROVED** [review]
Hi @fullstop125 and @samiullah997 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️  but you are almost there! 🏃🏾 

## Highlights
 - [x] Good job overall following the template design 💯 
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them, in order to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cwaku** [comment]
Kindly ignore the approval notice, my honest mistake. Please go ahead and resolve the changes and resubmit.
@fullstop125 and @samiullah997

File: script.js (hunk 1 above)
Comment (fullstop125) [reply script.js]: solved thanks

File: script.js (hunk 2 above)
Comment (fullstop125) [reply script.js]: solved thanks

**Review by Ugboaja-Uchechi — APPROVED** [review]
Hi @fullstop125 , 

## Status: APPROVED 🥳 🥳 

Great job working on the reviews suggested by the previous code reviewer 👍 . Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 
- Great job with your design and functionalities 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**